misc/hawkbit-upload.py

```diff
@@ -776,8 +777,118 @@ def add_or_update_artifact(self, file_name: str, module_id: str = None):
         )
         self.id["artifact"] = response["id"]
         return self.id["artifact"]
-
-
+    
+    def push_new_os_softwaremodule(
+        self,
+        name,
+    ):
+        # validate that both name and version does not exist
+        existing_module = self.get_softwaremodule_by_name(name, version=self.version)
+        if existing_module:
+            print(f"module {name} with version {self.version} already exists, using that")
+            self.id["newSoftwaremodule"] = response[0]['id']
+            return existing_module['id']
+        
+        data = [
+            {
+                "name": name,
+                "version": str(self.version),
+                "type": "os"
+            }
+        ]
+        
+        response = self.post("softwaremodules", data)
+        self.id["newSoftwaremodule"] = response[0]['id']
+        return self.id["newSoftwaremodule"]
+        
+    def push_new_distribution_set_with_os(
+        self,
+        distribution_name: str,
+        softwaremodule_name: str,
+        description: str = "",
+        module_ids: list = [],
+        dist_type: str = "os",
+        os_bundle_name: str = None
+    ): 
+        # Create the corresponding software "OS" Module
+        new_module_id = self.push_new_os_softwaremodule(softwaremodule_name)
+        
+        #define the corresponding data for the distribution set
+        assert isinstance(module_ids, list)
```

Comment:
Won't this cause a unrecoverable failure? What is our recovery case? Re-reubning?